documentation/modules/auxiliary/scanner/ssh/ssh_enumusers.md

```diff
@@ -0,0 +1,79 @@
+## Intro
+
+This module uses a malformed packet or timing attack to enumerate users on
+an OpenSSH server.
+
+Testing note: invalid users were logged, while valid users were not. YMMV.
+
+## Actions
+
+**Malformed Packet**
+
+The default action sends a malformed (corrupted) `SSH_MSG_USERAUTH_REQUEST`
+packet using public key authentication (must be enabled) to enumerate users.
+
+**Timing Attack**
+
+On some versions of OpenSSH under some configurations, OpenSSH will return a
+"permission denied" error for an invalid user faster than for a valid user,
+creating an opportunity for a timing attack to enumerate users.
+
+## Options
+
+**USERNAME**
+
+Single username to test (username spray).
+
+**USER_FILE**
+
+File containing usernames, one per line.
+
+**THRESHOLD**
+
+Amount of seconds needed before a user is considered found (timing attack only).
+
+**CHECK_FALSE**
+
+Check for false positives (random username).
+
+## Usage
+
+```
+msf5 > use auxiliary/scanner/ssh/ssh_enumusers
+msf5 auxiliary(scanner/ssh/ssh_enumusers) > set rhosts [redacted]
+rhosts => [redacted]
+msf5 auxiliary(scanner/ssh/ssh_enumusers) > echo $'wvu\nbcook' > users
```

Comment:
i like this trickyness for doc purproses

Reply:
It's laziness that still shows what I'm doing. :P

Reply:
Mr. One-liner